contracts/reputationMiningCycle/ReputationMiningCycleCommon.sol

```diff
@@ -31,6 +32,20 @@ contract ReputationMiningCycleCommon is ReputationMiningCycleStorage, PatriciaTr
   // in reputationMiningCycleRespond. If you change one, you should change the other.
   uint256 constant MINING_WINDOW_SIZE = 60 * 60 * 1; // 1 hour
 
+  function getMinerAddressIfStaked() internal view returns (address) {
+    // Is msg.sender a miner themselves? See if they have stake.
+    uint256 stakedForMining = IColonyNetwork(colonyNetworkAddress).getMiningStake(msg.sender).amount;
+    if (stakedForMining > 0) {
+      // If so, they we don't let them mine on someone else's behalf
+      return msg.sender;
+    }
+
+    // Return any delegator they are acting on behalf of
+    address delegator = IColonyNetwork(colonyNetworkAddress).getMiningDelegator(msg.sender);
```

Comment:
Should this be `msgSender()`? Also elsewhere in the function.

Reply:
The reputation mining contract does not support metatransactions as it's not a contract that a beginning user will be interacting with as they struggle to get xDAI to pay for gas (which is the reason for metatransactions, see #962)